packages/firestore/src/core/query.ts

```diff
@@ -37,6 +43,9 @@ export class Query {
   private memoizedCanonicalId: string | null = null;
   private memoizedOrderBy: OrderBy[] | null = null;
 
+  // The corresponding `Target` this `Query` instance.
+  private target: Target | null = null;
```

Comment:
Is there ever a case where `toTarget()` is not called? If there isn't, we can probably assign `this.target` in the constructor.

Reply:
Yes. `LocalStore.executeQuery` is one example.

More importantly, `Query` is built in a way where there might be many temporary instances, every `with*` will create one for instance.